packages/realm/src/app-services/SyncSession.ts

```diff
@@ -53,13 +53,29 @@ export enum ProgressMode {
   ForCurrentlyOutstandingWork = "forCurrentlyOutstandingWork",
 }
 
-export type ProgressNotificationCallback =
+/** @deprecated */
+export type PartitionBasedSyncProgressNotificationCallback =
   /**
+   * This notification callback only supports Partition Based Sync.
    * @param transferred - The current number of bytes already transferred
    * @param transferable - The total number of transferable bytes (i.e. the number of bytes already transferred plus the number of bytes pending transfer)
+   * @deprecated - This notification callback will be removed in next major release. Use {@link DynamicProgressNotificationCallback} instead.
+   * @since 1.12.0
    */
   (transferred: number, transferable: number) => void;
 
+export type DynamicProgressNotificationCallback =
+  /**
+   * This notification callback supports both Partition Based Sync and Flexible Sync.
+   * @param estimate - An estimate between 0.0 and 1.0 of how much have been transferred.
+   * @since 12.10.0
```

Comment:
`@since` should be updated (or removed 🤔 ) @kneth , when do we usually add `@since`?

Reply:
@kneth, I removed the `@since` annotation for now.